xds/src/main/java/io/grpc/xds/orca/OrcaMetricReportingServerInterceptor.java

```diff
@@ -94,8 +111,42 @@ private static OrcaLoadReport fromInternalReport(MetricReport internalReport) {
     return OrcaLoadReport.newBuilder()
         .setCpuUtilization(internalReport.getCpuUtilization())
         .setMemUtilization(internalReport.getMemoryUtilization())
+        .setRpsFractional(internalReport.getQps())
         .putAllUtilization(internalReport.getUtilizationMetrics())
         .putAllRequestCost(internalReport.getRequestCostMetrics())
         .build();
   }
+
+  /**
+   * Return a merged {@link OrcaLoadReport} where the metrics from {@link CallMetricRecorder} takes
+   * a higher precedence compared to {@link MetricRecorder}.
+   */
+  private static OrcaLoadReport mergeMetrics(OrcaLoadReport callMetricRecorderReport,
+      OrcaLoadReport metricRecorderReport) {
+    // Merge metrics from the MetricRecorder first since metrics from the CallMetricRecorder takes a
+    // higher precedence.
+    OrcaLoadReport.Builder builder = metricRecorderReport.toBuilder()
```

Comment:
I'd have expected simply:
```java
metricRecorderReport.toBuilder().mergeFrom(callMetricRecorderReport).build();
```

And combining that with changing `fromInternalReport()` to return `OrcaLoadReport.Builder`, I think you can avoid two copies (one build() and this toBuilder()). (You could even pass in a Builder instance to `fromInternalReport()`, and then conditionally set fields, to remove two more copies.)

The current code here looks pretty broken for utilization, as it discards the metricRecorderReport's data entirely.

Reply:
Thanks, C++ currently replaces the whole map but I think this is what we intended with the design.

The only corner case I am concerned is when the user calls SetAllUtilization() in the per-server recorder. I assume the user will expect them to be atomic but we may only include a subset of them. Atomicity is not guaranteed for metrics recorded individually (pending gRFC A51 mentions this) so it's not an issue.

@ejona86 / @markdroth , should we then:
- Add to the A51 update that the "SetAllUtilization()" and (soon to be added) SetAllNamedMetric() APIs may get partially overridden by metrics recorded with the per-call API.
- Update the current C++ implementation to merge the map. This is tiny more expensive then today but I think will be fine.

LMK.

Reply:
Yeah, merging seems like the right thing to do here.  Given the data model here (server metric recorder and call metric recorder each store their own state, and fields set in call metric recorder take precedence), I would not expect users would think that calling `SetAllUtilization()` on either recorder will affect the values in the other recorder.

Reply:
Thanks, make sense. I'll make this change in C++.